src/sentry/api/endpoints/organization_integrations.py

```diff
@@ -17,6 +21,20 @@ def get(self, request, organization):
         if "provider_key" in request.GET:
             integrations = integrations.filter(integration__provider=request.GET["provider_key"])
 
+        # XXX(meredith): Filter out workspace apps if there are any.
+        if not features.has(
+            "organizations:slack-allow-workspace", organization=organization, actor=request.user
+        ):
+            slack_integrations = integrations.filter(integration__provider="slack")
+            workspace_ids = [
+                workspace_app.id
+                for workspace_app in filter(
+                    lambda i: get_integration_type(i.integration) == "workspace_app",
+                    slack_integrations,
+                )
+            ]
+            integrations = integrations.filter(id__in=workspace_ids)
```

Comment:
Can you add a test that makes sure the workspace apps are missing when the flag is off?

Reply:
@mgaeta just added/updated tests in `test_notify_action`. It's a bit odd though because we have tests in https://github.com/getsentry/sentry/blob/master/tests/sentry/integrations/slack/test_utils.py#L25 as well. I added a test to check that bot apps worked (since all of the ones currently were just testing workspace) and then a couple others to test invalid form and preventing the notification from being sent for workspace apps

Reply:
I was thinking a test in `tests/sentry/api/endpoints/test_organization_integrations.py` to make sure that this block correctly filters.